### PR TITLE
Enhancement: Removing mutable params, adding type hints; #5169

### PR DESCRIPTION
**Pull request by erlingstaff**
Issue: #5169 
This PR serves two purposes:

1. Removes all instances of mutable default parameters (e.g. rses=[], filters={})
2. Type hints all the parameters in the functions where the mutable parameters appear. #6588

### CODE REVIEW

**Review by rdimaio — COMMENTED** [review]
Most of the changes in the commit do not just remove mutable parameters, but add type annotations as well. 

In principle these should be two changes done in separate commits, but now that they're already grouped together I can understand that it would be an absolute pain to split the commit into two (one for type annotations and one for the mutable parameters), as both changes are on the same line, so `git add -p` wouldn't be enough. 

I am personally fine with having them both in the same commit in this instance, but in future it's best to only have one "type" of change per commit.

Still, I think that the commit and PR titles should be changed to also reflect the fact that type annotations are being added (and it would be great if you could link to the type annotations issue in the PR description, for reference: https://github.com/rucio/rucio/issues/6588)

**Comment by erlingstaff** [comment]
> Most of the changes in the commit do not just remove mutable parameters, but add type annotations as well.
> 
> In principle these should be two changes done in separate commits,

Good to know, thanks. Had a conversation with @bari12 and he allowed this one PR to slip by as a single commit


> Still, I think that the commit and PR titles should be changed to also reflect the fact that type annotations are being added (and it would be great if you could link to the type annotations issue in the PR description, for reference: #6588)

Done

File: lib/rucio/rse/protocols/webdav.py [hunk 1]
```diff
@@ -184,12 +185,15 @@ def connect(self, credentials={}):
...
+                    self.cert = (proxy_path, proxy_path)
                 elif self.auth_token:
+                    # If no proxy is found, we set the cert to None and use the auth_token
+                    self.cert = None
                     pass
                 else:
                     raise exception.RSEAccessDenied('X509_USER_PROXY is not set')
-            self.cert = (x509, x509)
+            else:
+                self.cert = (x509, x509)
```
Comment (rdimaio) [inline lib/rucio/rse/protocols/webdav.py:196]: Could you separate this into a separate commit? It's a functional change that's unrelated to mutable params and type hints, and it's probably best to have it separately. You can reset the commit and `git add -p lib/rucio/rse/protocols/webdav.py` to only select these changes.

File: lib/rucio/api/did.py [hunk 2]
```diff
@@ -66,7 +66,22 @@ def list_dids(scope, filters, did_type='collection', ignore_case=False, limit=No
...
-def add_did(scope, name, did_type, issuer, account=None, statuses={}, meta={}, rules=[], lifetime=None, dids=[], rse=None, vo='def', *, session: "Session"):
+def add_did(
+    scope: str,
+    name: str,
+    did_type: str,
+    issuer: str,
+    account: "Optional[str]" = None,
+    statuses: "Optional[dict[str, str]]" = None,
+    meta: "Optional[dict[str, str]]" = None,
+    rules: "Optional[list[dict[str, Any]]]" = None,
```
Comment (rdimaio) [inline lib/rucio/api/did.py]: `Sequence` instead of `list` to match the existing `add_did` signature in `core`: https://github.com/rucio/rucio/blob/19392c29c61021d37aa1f3bbe6017722222994cf/lib/rucio/core/did.py#L119-L131

File: lib/rucio/api/did.py [hunk 3]
```diff
@@ -66,7 +66,22 @@ def list_dids(scope, filters, did_type='collection', ignore_case=False, limit=No
...
+    lifetime: "Optional[str]" = None,
+    dids: "Optional[list[dict[str, Any]]]" = None,
```
Comment (rdimaio) [inline lib/rucio/api/did.py]: `Sequence` instead of `list` to match the existing `add_did` signature in `core`: https://github.com/rucio/rucio/blob/19392c29c61021d37aa1f3bbe6017722222994cf/lib/rucio/core/did.py#L119-L131

File: lib/rucio/api/account.py [hunk 4]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any, Iterator, Optional
```
Comment (rdimaio) [inline lib/rucio/api/account.py]: `Iterator` should be imported from `collections.abc` (see https://docs.python.org/3/library/typing.html#typing.Iterator)

File: lib/rucio/api/did.py [hunk 5]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 from copy import deepcopy
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any, Iterator
```
Comment (rdimaio) [inline lib/rucio/api/did.py]: `Iterator` should be imported from `collections.abc` (see https://docs.python.org/3/library/typing.html#typing.Iterator)

File: lib/rucio/api/replica.py [hunk 6]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 import datetime
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any, Iterator, Optional
```
Comment (rdimaio) [inline lib/rucio/api/replica.py]: `Iterator` should be imported from `collections.abc` (see https://docs.python.org/3/library/typing.html#typing.Iterator)

File: lib/rucio/api/rule.py [hunk 7]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, Iterator, Optional
```
Comment (rdimaio) [inline lib/rucio/api/rule.py]: `Iterator` should be imported from `collections.abc` (see https://docs.python.org/3/library/typing.html#typing.Iterator)

File: lib/rucio/core/did_meta_plugins/filter_engine.py [hunk 8]
```diff
@@ -17,10 +17,11 @@
 import operator
 from datetime import date, datetime, timedelta
 from importlib import import_module
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any, Callable, Iterable, Optional
```
Comment (rdimaio) [inline lib/rucio/core/did_meta_plugins/filter_engine.py]: `Iterable` should be imported from `collections.abc` (see: https://docs.python.org/3/library/typing.html#typing.Iterable)

File: lib/rucio/core/rse_selector.py [hunk 9]
```diff
@@ -13,9 +13,10 @@
 # limitations under the License.
 
 from random import shuffle, uniform
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Iterable, Optional
```
Comment (rdimaio) [inline lib/rucio/core/rse_selector.py]: `Iterable` should be imported from `collections.abc` (see: https://docs.python.org/3/library/typing.html#typing.Iterable)

File: lib/rucio/core/did_meta_plugins/filter_engine.py [hunk 10]
```diff
@@ -271,13 +272,17 @@ def _try_typecast_string(self, value):
             pass
         return value
 
-    def create_mongo_query(self, additional_filters={}):
+    def create_mongo_query(
+        self,
+        additional_filters: Optional[list[tuple[str, Callable, str]]] = None
```
Comment (rdimaio) [inline lib/rucio/core/did_meta_plugins/filter_engine.py]: `Iterable` instead of `list` (`additional_filters` is only used in a for loop)

File: lib/rucio/core/did_meta_plugins/filter_engine.py [hunk 11]
```diff
@@ -326,15 +331,20 @@ def create_mongo_query(self, additional_filters={}):
 
         return query_str
 
-    def create_postgres_query(self, additional_filters={}, fixed_table_columns=('scope', 'name', 'vo'),
-                              jsonb_column='data'):
+    def create_postgres_query(
+        self,
+        additional_filters: Optional[list[tuple[str, Callable, str]]] = None,
```
Comment (rdimaio) [inline lib/rucio/core/did_meta_plugins/filter_engine.py]: `Iterable` instead of `list`

File: lib/rucio/core/did_meta_plugins/filter_engine.py [hunk 12]
```diff
@@ -400,7 +410,14 @@ def create_postgres_query(self, additional_filters={}, fixed_table_columns=('sco
         return ' OR '.join(or_expressions)
 
     @read_session
-    def create_sqla_query(self, *, session: "Session", additional_model_attributes=[], additional_filters={}, json_column=None):
+    def create_sqla_query(
+        self,
+        *,
+        session: "Session",
+        additional_model_attributes: Optional[list[InstrumentedAttribute]] = None,
+        additional_filters: Optional[list[tuple[str, Callable, str]]] = None,
```
Comment (rdimaio) [inline lib/rucio/core/did_meta_plugins/filter_engine.py]: `Iterable` instead of `list`

File: lib/rucio/core/replica.py [hunk 13]
```diff
@@ -1600,7 +1601,22 @@ def _expected_pfns(lfns, rse_settings, scheme, operation='write', domain='wan',
...
+    adler32: "Optional[str]" = None,
+    md5: "Optional[str]" = None,
+    dsn: "Optional[str]" = None,
+    pfn: "Optional[str]" = None,
+    meta: "Optional[dict[str, Any]]" = None,
+    rules: "Optional[list[dict[str, Any]]]" = None,
+    tombstone: "Optional[datetime]" = None,
+    *,
+    session: "Session"
+) -> bool:
```
Comment (rdimaio) [inline lib/rucio/core/replica.py]: This actually returns a `list[dict]`. 

`add_replica` returns `add_replicas`: https://github.com/rucio/rucio/blob/19392c29c61021d37aa1f3bbe6017722222994cf/lib/rucio/core/replica.py#L1644

Which returns `replicas`: https://github.com/rucio/rucio/blob/19392c29c61021d37aa1f3bbe6017722222994cf/lib/rucio/core/replica.py#L1600

Which is defined here: https://github.com/rucio/rucio/blob/19392c29c61021d37aa1f3bbe6017722222994cf/lib/rucio/core/replica.py#L1562

Which returns this list:

https://github.com/rucio/rucio/blob/19392c29c61021d37aa1f3bbe6017722222994cf/lib/rucio/core/replica.py#L1453

So it's a case of the existing docstrings being wrong, as they say that all these functions return a `bool`, which is not the case

File: lib/rucio/core/replica.py [hunk 14]
```diff
@@ -3426,7 +3442,19 @@ def get_replicas_state(scope=None, name=None, *, session: "Session"):
 
 
 @read_session
-def get_suspicious_files(rse_expression, available_elsewhere, filter_=None, logger=logging.log, younger_than=5, nattempts=0, nattempts_exact=False, *, session: "Session", exclude_states=['B', 'R', 'D'], is_suspicious=False):
+def get_suspicious_files(
+    rse_expression: str,
+    available_elsewhere: int,
```
Comment (rdimaio) [inline lib/rucio/core/replica.py:3448]: In theory this could be more precisely set as `SuspiciousAvailability`, but the check on line 3484 expands the `SuspiciousAvailability` enum into a `list[int]`, so it expects an `int`. No change is needed here, maybe in future we can think of modifying the change on line 3484 so that we can annotate `available_elsewhere` as `SuspiciousAvailability`.

File: lib/rucio/core/replica.py [hunk 15]
```diff
@@ -3426,7 +3442,19 @@ def get_replicas_state(scope=None, name=None, *, session: "Session"):
...
+    filter_: "Optional[dict[str, Any]]" = None,
+    logger: "LoggerFunction" = logging.log,
+    younger_than: "Optional[datetime]" = None,
+    nattempts: int = 0,
+    nattempts_exact: bool = False,
+    *,
+    session: "Session",
+    exclude_states: "Optional[list[str]]" = None,
```
Comment (rdimaio) [inline lib/rucio/core/replica.py]: `Iterable` instead of `list`

File: lib/rucio/core/rse_selector.py [hunk 16]
```diff
@@ -245,12 +255,27 @@ def __choose_rse(self, rses):
...
-def resolve_rse_expression(rse_expression, account, weight=None, copies=1, ignore_account_limit=False, size=0, preferred_rses=[], blocklist=[], prioritize_order_over_weight=False, existing_rse_size=None, *, session: "Session"):
+def resolve_rse_expression(
+    rse_expression: str,
+    account: InternalAccount,
+    weight: Optional[int] = None,
+    copies: int = 1,
+    ignore_account_limit: bool = False,
+    size: int = 0,
+    preferred_rses: Optional[list[str]] = None,
+    blocklist: Optional[list[str]] = None,
```
Comment (rdimaio) [inline lib/rucio/core/rse_selector.py]: `Sequence` instead of `str` (technically I think `Collection` would be better here, but we're not using the `Collection` type hint anywhere in the code at the moment, so I think it might be best to avoid it for now)

File: lib/rucio/core/rse_selector.py [hunk 17]
```diff
@@ -133,7 +134,15 @@ def __init__(self, account, rses, weight, copies, ignore_account_limit=False, *,
         # don't consider removing rses based on the total space here - because files already on the RSE are taken into account
         # it is possible to have no space but still be able to fulfil the rule
 
-    def select_rse(self, size, preferred_rse_ids, copies=0, blocklist=[], prioritize_order_over_weight=False, existing_rse_size=None):
+    def select_rse(
+        self,
+        size: int,
+        preferred_rse_ids: Iterable[str],
+        copies: int = 0,
+        blocklist: Optional[list[str]] = None,
```
Comment (rdimaio) [inline lib/rucio/core/rse_selector.py]: `Sequence` instead of `str`

File: lib/rucio/core/rse_selector.py [hunk 18]
```diff
@@ -245,12 +255,27 @@ def __choose_rse(self, rses):
...
 @read_session
-def resolve_rse_expression(rse_expression, account, weight=None, copies=1, ignore_account_limit=False, size=0, preferred_rses=[], blocklist=[], prioritize_order_over_weight=False, existing_rse_size=None, *, session: "Session"):
+def resolve_rse_expression(
+    rse_expression: str,
+    account: InternalAccount,
+    weight: Optional[int] = None,
+    copies: int = 1,
+    ignore_account_limit: bool = False,
+    size: int = 0,
+    preferred_rses: Optional[list[str]] = None,
```
Comment (rdimaio) [inline lib/rucio/core/rse_selector.py]: `Iterable` instead of `list`

File: lib/rucio/core/rule_grouping.py [hunk 19]
```diff
@@ -223,7 +226,20 @@ def create_transfer_dict(dest_rse_id, request_type, scope, name, rule, lock=None
...
 @transactional_session
-def __apply_rule_to_files_none_grouping(datasetfiles, locks, replicas, source_replicas, rseselector, rule, preferred_rse_ids=[], source_rses=[], *, session: "Session"):
+def __apply_rule_to_files_none_grouping(
+    datasetfiles: Sequence[dict[str, Any]],
+    locks: dict[tuple[InternalScope, str], Sequence[models.ReplicaLock]],
+    replicas: dict[tuple[InternalScope, str], Sequence[models.CollectionReplica]],
+    source_replicas: dict[tuple[InternalScope, str], Sequence[models.CollectionReplica]],
+    rseselector: RSESelector,
+    rule: models.ReplicationRule,
+    preferred_rse_ids: Optional[list[str]] = None,
```
Comment (rdimaio) [inline lib/rucio/core/rule_grouping.py]: `Sequence` instead of `list`

File: lib/rucio/core/rule_grouping.py [hunk 20]
```diff
@@ -223,7 +226,20 @@ def create_transfer_dict(dest_rse_id, request_type, scope, name, rule, lock=None
...
+    source_rses: Optional[list[str]] = None,
```
Comment (rdimaio) [inline lib/rucio/core/rule_grouping.py]: `Sequence` instead of `list`

File: lib/rucio/core/rule_grouping.py [hunk 21]
```diff
@@ -304,7 +322,20 @@ def __apply_rule_to_files_none_grouping(datasetfiles, locks, replicas, source_re
...
 @transactional_session
-def __apply_rule_to_files_all_grouping(datasetfiles, locks, replicas, source_replicas, rseselector, rule, preferred_rse_ids=[], source_rses=[], *, session: "Session"):
+def __apply_rule_to_files_all_grouping(
+    datasetfiles: Sequence[dict[str, Any]],
+    locks: dict[tuple[InternalScope, str], Sequence[models.ReplicaLock]],
+    replicas: dict[tuple[InternalScope, str], Sequence[models.CollectionReplica]],
+    source_replicas: dict[tuple[InternalScope, str], Sequence[models.CollectionReplica]],
+    rseselector: RSESelector,
+    rule: models.ReplicationRule,
+    preferred_rse_ids: Optional[list[str]] = None,
```
Comment (rdimaio) [inline lib/rucio/core/rule_grouping.py]: `Sequence` instead of `list`

File: lib/rucio/core/rule_grouping.py [hunk 22]
```diff
@@ -304,7 +322,20 @@ def __apply_rule_to_files_none_grouping(datasetfiles, locks, replicas, source_re
...
+    source_rses: Optional[list[str]] = None,
```
Comment (rdimaio) [inline lib/rucio/core/rule_grouping.py]: `Sequence` instead of `list`

File: lib/rucio/core/rule_grouping.py [hunk 23]
```diff
@@ -424,7 +457,20 @@ def __apply_rule_to_files_all_grouping(datasetfiles, locks, replicas, source_rep
...
 @transactional_session
-def __apply_rule_to_files_dataset_grouping(datasetfiles, locks, replicas, source_replicas, rseselector, rule, preferred_rse_ids=[], source_rses=[], *, session: "Session"):
+def __apply_rule_to_files_dataset_grouping(
+    datasetfiles: Sequence[dict[str, Any]],
+    locks: dict[tuple[InternalScope, str], Sequence[models.ReplicaLock]],
+    replicas: dict[tuple[InternalScope, str], Sequence[models.CollectionReplica]],
+    source_replicas: dict[tuple[InternalScope, str], Sequence[models.CollectionReplica]],
+    rseselector: RSESelector,
+    rule: models.ReplicationRule,
+    preferred_rse_ids: Optional[list[str]] = None,
```
Comment (rdimaio) [inline lib/rucio/core/rule_grouping.py]: `Sequence` instead of `list`

File: lib/rucio/core/rule_grouping.py [hunk 24]
```diff
@@ -424,7 +457,20 @@ def __apply_rule_to_files_all_grouping(datasetfiles, locks, replicas, source_rep
...
+    source_rses: Optional[list[str]] = None,
```
Comment (rdimaio) [inline lib/rucio/core/rule_grouping.py]: `Sequence` instead of `list`

File: lib/rucio/web/ui/flask/common/utils.py [hunk 25]
```diff
@@ -100,7 +101,14 @@ def prepare_saml_request(environ, data):
     return None
 
 
-def add_cookies(response, cookie={}):
+def add_cookies(response: Response, cookie: Optional[list[dict[str, Any]]] = None) -> Response:
```
Comment (rdimaio) [inline lib/rucio/web/ui/flask/common/utils.py]: `Iterable` instead of `list`

File: lib/rucio/core/rse_selector.py (hunk 16 above)
Comment (erlingstaff) [reply lib/rucio/core/rse_selector.py]: Do you mean `Sequence[str]` instead of `list[str]`? In this particular case blocklist is passed from `daemons/transmogrifier.py` as:

```
blocklisted_rse_id = [rse["id"] for rse in list_rses({"availability_write": False})]
```

Comment (rdimaio) [reply lib/rucio/core/rse_selector.py]: Yeah, sorry, I meant `Sequence[str]` instead of `list[str]`

File: lib/rucio/rse/protocols/webdav.py (hunk 1 above)
Comment (erlingstaff) [reply lib/rucio/rse/protocols/webdav.py:196]: Moved into a separate commit, 8e776054283ab2937c6ba7c10d25e10566a1101d

File: lib/rucio/api/did.py (hunk 2 above)
Comment (erlingstaff) [reply lib/rucio/api/did.py]: Changed, now: 
```
rules: "Optional[Sequence[dict[str, Any]]]" = None,
```

File: lib/rucio/api/did.py (hunk 3 above)
Comment (erlingstaff) [reply lib/rucio/api/did.py]: Changed, now: 
```
dids: "Optional[Sequence[dict[str, Any]]]" = None,
```

File: lib/rucio/api/account.py (hunk 4 above)
Comment (erlingstaff) [reply lib/rucio/api/account.py]: Importing `Iterator` from `collections.abc` now

File: lib/rucio/api/did.py (hunk 5 above)
Comment (erlingstaff) [reply lib/rucio/api/did.py]: Importing `Iterator` from `collections.abc` now

File: lib/rucio/api/replica.py (hunk 6 above)
Comment (erlingstaff) [reply lib/rucio/api/replica.py]: Importing `Iterator` from `collections.abc` now

File: lib/rucio/api/rule.py (hunk 7 above)
Comment (erlingstaff) [reply lib/rucio/api/rule.py]: Importing `Iterator` from `collections.abc` now

File: lib/rucio/core/did_meta_plugins/filter_engine.py (hunk 8 above)
Comment (erlingstaff) [reply lib/rucio/core/did_meta_plugins/filter_engine.py]: Importing `Iterable`, and, as discussed in private messages, `Callable` from collections.abc now

File: lib/rucio/core/rse_selector.py (hunk 9 above)
Comment (erlingstaff) [reply lib/rucio/core/rse_selector.py]: Importing `Iterable` from `collections.abc` now

File: lib/rucio/core/did_meta_plugins/filter_engine.py (hunk 10 above)
Comment (erlingstaff) [reply lib/rucio/core/did_meta_plugins/filter_engine.py]: Changed, now

```
additional_filters: Optional[Iterable[tuple[str, Callable, str]]] = None
```

File: lib/rucio/core/did_meta_plugins/filter_engine.py (hunk 11 above)
Comment (erlingstaff) [reply lib/rucio/core/did_meta_plugins/filter_engine.py]: Changed, now:
```
additional_filters: Optional[Iterable[tuple[str, Callable, str]]] = None
```

File: lib/rucio/core/did_meta_plugins/filter_engine.py (hunk 12 above)
Comment (erlingstaff) [reply lib/rucio/core/did_meta_plugins/filter_engine.py]: Changed, now:
```
additional_filters: Optional[Iterable[tuple[str, Callable, str]]] = None,
```

File: lib/rucio/core/replica.py (hunk 13 above)
Comment (erlingstaff) [reply lib/rucio/core/replica.py]: Changed this to return:
```
list[dict[str, Any]
```

I also updated the docstrings for those three functions (add_replicas, __bulk_add_file_dids, and add_replica) to properly convey that they return a list of replicas. This was done in a separate commit: 0cc29a202759180a8c8ec0231c5f51d63e9b7be5

File: lib/rucio/core/replica.py (hunk 15 above)
Comment (erlingstaff) [reply lib/rucio/core/replica.py]: Changed to:
```
exclude_states: "Optional[Iterable[str]]" = None,
```

File: lib/rucio/core/rse_selector.py (hunk 16 above)
Comment (erlingstaff) [reply lib/rucio/core/rse_selector.py]: Changed to:

```
blocklist: Optional[Sequence[str]] = None,
```

File: lib/rucio/core/rse_selector.py (hunk 17 above)
Comment (erlingstaff) [reply lib/rucio/core/rse_selector.py]: (Assuming you mean `Sequence` instead of `list` here as well)

Changed to:
```
blocklist: Optional[Sequence[str]] = None,
```

File: lib/rucio/core/rse_selector.py (hunk 18 above)
Comment (erlingstaff) [reply lib/rucio/core/rse_selector.py]: Changed to 
```
preferred_rses: Optional[Iterable[str]] = None,
```

File: lib/rucio/core/rule_grouping.py (hunk 19 above)
Comment (erlingstaff) [reply lib/rucio/core/rule_grouping.py]: Changed to:

```
    preferred_rse_ids: Optional[Sequence[str]] = None,
```

File: lib/rucio/core/rule_grouping.py (hunk 20 above)
Comment (erlingstaff) [reply lib/rucio/core/rule_grouping.py]: Changed to:

```
source_rses: Optional[Sequence[str]] = None,
```

File: lib/rucio/core/rule_grouping.py (hunk 21 above)
Comment (erlingstaff) [reply lib/rucio/core/rule_grouping.py]: Changed to:

```
preferred_rse_ids: Optional[Sequence[str]] = None,
```

File: lib/rucio/core/rule_grouping.py (hunk 22 above)
Comment (erlingstaff) [reply lib/rucio/core/rule_grouping.py]: Changed to:

```
source_rses: Optional[Sequence[str]] = None,
```

File: lib/rucio/core/rule_grouping.py (hunk 23 above)
Comment (erlingstaff) [reply lib/rucio/core/rule_grouping.py]: Changed to:

```
preferred_rse_ids: Optional[Sequence[str]] = None,
```

File: lib/rucio/core/rule_grouping.py (hunk 24 above)
Comment (erlingstaff) [reply lib/rucio/core/rule_grouping.py]: Changed to:

```
source_rses: Optional[Sequence[str]] = None,
```

File: lib/rucio/web/ui/flask/common/utils.py (hunk 25 above)
Comment (erlingstaff) [reply lib/rucio/web/ui/flask/common/utils.py]: Changed to:

```
def add_cookies(response: Response, cookie: Optional[Iterable[dict[str, Any]]] = None) -> Response:
```

**Review by rdimaio — DISMISSED** [review]

**Comment by bari12** [comment]
Please rebase and resolve conflict :-)

**Comment by erlingstaff** [comment]
> Please rebase and resolve conflict :-)

Done 👍

**Review by rdimaio — APPROVED** [review]
